features/cellular/framework/targets/UBLOX/N2XX/UBLOX_N2XX_CellularSIM.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2019, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "UBLOX_N2XX_CellularSIM.h"
+#include "CellularLog.h"
+#include "platform/mbed_wait_api.h"
+
+using namespace mbed;
+
+UBLOX_N2XX_CellularSIM::UBLOX_N2XX_CellularSIM(ATHandler &atHandler) : AT_CellularSIM(atHandler)
+{
+    _at.set_urc_handler("+NPIN:", callback(this, &UBLOX_N2XX_CellularSIM::NPIN_URC));
+    memset(simstr,0,sizeof(simstr));
+}
+
+UBLOX_N2XX_CellularSIM::~UBLOX_N2XX_CellularSIM()
+{
+}
+
+// Callback for Sim Pin.
+void UBLOX_N2XX_CellularSIM::NPIN_URC()
+{
+    _at.read_string(simstr, sizeof(simstr));
+}
+
+nsapi_error_t UBLOX_N2XX_CellularSIM::get_sim_state(SimState &state)
```

Comment:
Move to `UBLOX_N2XX_CellularDevice`.

Reply:
`UBLOX_N2XX` is already inheriting form `AT_CellularDevice` then why need to create `UBLOX_N2XX_CellularDevice`.

Reply:
My bad, you are right and it should be ´virtual nsapi_error_t `UBLOX_N2XX::get_sim_state(SimState &state)`

Reply:
Alright, No problem.